backend/ohq/permissions.py

```diff
@@ -503,3 +503,107 @@ def has_permission(self, request, view):
             return True
 
         return True
+
+class DocumentCreatePermission(permissions.BasePermission):
+    def has_permission(self, request, view):
+        if not request.user.is_authenticated:
+            return False
+
+        membership = Membership.objects.filter(
+            course=view.kwargs["course_pk"], user=request.user
+        ).first()
+
+        # Non-Students can't do anything
+        if membership is None:
+            return False
+
+        if view.action == "retrieve":
+            return membership.is_ta
+
+        # Head TAs+ can make changes
+        if view.action in ["retrieve", "create", "destroy", "update", "partial_update"]:
```

Comment:
You can remove the "retrieve" from the list of allowed operations here if the behavior for retrieve is already defined like above!